### ISSUE TITLE
(Enhancement) Add support for PushBullet notifications

### ISSUE THREAD

**Issue by jahknee** [PokeAlert/PokeAlert#235]
Instead of limiting notifications to the device used for scanning, with PushBullet the users could choose if they want to get the notifications on their computer, iPad or any device where they have installed PushBullet. They could run PokeAlert on an Android device, or in Bluestacks in a virtual machine over VPN, and still get notifications to their main device. There would be no need for bringing a second device with you in order to not miss any notifications.

By using a notification of the type "link", you can also attach a google maps link in the notification. When the user taps the notification, the position is shown in google maps.

The implementation is _very_ easy, I have myself implemented PushBullet  in a .NET application that was used for scanning and it sent notifications to multiple PushBullet accounts. It didn't take many hours to get it working and added a whole lot of value.

Users should be made aware of that the free version of PushBullet is limited to 500 notifications per month. This is however more than enough if you only care about the rare Pokemons. And if the user should run out of notifications, they could simply get a second PushBullet account. 

[Create push documentaton](https://docs.pushbullet.com/#create-push)

**Comment by vlacorte** [comment]
What are the advantages of PushBullet over redirecting the notification to Telegram (via a bot, e.g. to notify multiple subscribers)?

**Comment by jahknee** [comment]
I'm not entirely sure who does what in your mentioned solution (bot + telegram). Does this involve users creating and using a bot? Are there easy ready-to-use solutions? I found this at https://core.telegram.org/bots/faq:

_"Creating Telegram bots is super-easy, but you will need at least some skills at computer programming. In order for a bot to work, set up a bot account with @BotFather, then connect it to your backend server via our API."_

While this probably is no bigger problem for the more tech-savvy users, this is ofcourse a bigger hurdle for the average pokemon go player.

Since i don't know much about the solution you mentioned, I can't make any comparisons between the two. But I like PushBullet because of the following reasons:

- a RESTful API which is very easy to use
- Also easy for users to get up and running. Install PushBullet on device(s), login on PushBullet web page and generate API key, paste API key in the consuming app. Done.

**Comment by vlacorte** [comment]
@jahknee The Telegram bot is a utility to allow notifications to be pushed into a group/conversation (among many other functionalities, but let's focus on this one) since it can't be programmed to notify Telegram users (since that would allow spamming). It seems that both approaches are very similar but, as you have stated, PushBullet seems more straightforward.

Nonetheless, they both would add another layer of complexity and the average pogo player probably wouldn't bother with either. Following this line, we should focus on implementing the most straightforward and fastest since it feels unlikely that the current hash algorithms are going to last too long.

**Comment by jahknee** [comment]
@vlacorte I agree that there might be users who won't be able to get even PushBullet running.

However, the benefits for the rest of the users are quite big. You won't have to drag along a second device with you. Activate PokeAlert on your device, connect it to the charger and leave home with your main device. I think many users would use PushBullet if it was available due to the benefits:

- Decrease battery drain.
- Decrease data usage (device is at home, using wifi)
- Get notifications to, for example, your desktop/laptop at work/home, not having to listen for notifications from phone.
- Share notifications amongst family and friends

All features adds complexity, but implementing support for PushBullet is a very small effort. The threat of hash algoritms changing will probably be a never ending threat, but that shouldn't stop developers from implementing more features.

I've stated my case now and I won't nag more about this. :)

**Comment by vlacorte** [comment]
@jahknee I completely agree with all your points, and that is why we ended in the same discussion thread; if, for some reason, you think otherwise than it's just miscommunication :)

Aside from stating the use cases, which are very important at the moment (and probably the only "flaw" of Pokealert), are you willing to start implementing the PushBullet hook? I don't have experience with Android/mobile programming but this minor context, redirecting notifications, shouldn't be prohibitively hard to grasp, so count me in.

**Comment by jahknee** [comment]
@vlacorte Ok, it seems I misunderstood you. :)

Unfortunately I have no experience with Android programming either. I'm a .NET developer and I'm working on updating my scanner (used FPM earlier) which will be more of a centralised (run scan on a computer, distribute notifications to users) scanner. I simply decided to share some ideas for features with PokeAlert that, in my eyes, would increase it's usability.
